### PR TITLE
[MCKIN-7755] [OC-4523] Incorporate completion api changes in api-integration v1 

### PR DESCRIPTION
**Pull request by xitij2000**
Bump version of api-integration and fix bug in module_render

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @xitij2000! I've created [OSPR-2492](https://openedx.atlassian.net/browse/OSPR-2492) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx-solutions/edx-platform/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

File: lms/djangoapps/courseware/module_render.py [hunk 1]
```diff
@@ -591,7 +591,7 @@ def handle_deprecated_progress_event(block, event):
             )
         else:
             if user_id != user.id:
-                log.warning("{} tried to submit a completion on behalf of {}".format(user, requested_user_id))
+                log.warning("{} tried to submit a completion on behalf of {}".format(user, user.id))
```
Comment (jcdyer) [inline lms/djangoapps/courseware/module_render.py]: This should be user_id, not user.id.  The point is that the user was trying to submit someone else's completion.  Returning their own ID doesn't convey that message.

**Comment by jcdyer** [comment]
I'm looking into the test failures.  The "missing" import is present at `cms/djangoapps/contentstore/views/item.py`, and I don't see anything obvious in your code that should break that.  If I had to guess, I'd suspect either circular import issues, or some issue arising from the module being imported from one place as `contentstore.views.item` and from another place as `cms.djangoapps.contentstore.views.item`.

**Comment by jcdyer** [comment]
I see some other test failures due to changed mongo query counts.  Those can usually just be adjusted in the test, but it's good to try to understand where to query change is coming from, and make sure that we haven't turned an O(1) set of queries into an O(N).

**Comment by jcdyer** [comment]
I need to step away and take care of some other work.  Be back in a little bit.

**Comment by xitij2000** [comment]
@jcdyer Forgot to comment here but the missing import issue goes away if I undo the change in: https://github.com/edx-solutions/course-edx-platform-extensions/pull/13/ however I don't know why that would be the case.

**Comment by jcdyer** [comment]
One oddity I've found is that the Aggregations are only going up to the chapter, not the course. In the first test failure, Just before requesting the aggregation data, I'm getting the following aggregators reported:

```python
 [<Aggregator: Aggregator: robot6, course-v1:org.0+course_0+Run_0, block-v1:org.0+course_0+Run_0+type@vertical+block@Vertical_Sequence: 0.0>,
  <Aggregator: Aggregator: robot6, course-v1:org.0+course_0+Run_0, block-v1:org.0+course_0+Run_0+type@sequential+block@Sequential: 0.0>,
  <Aggregator: Aggregator: robot6, course-v1:org.0+course_0+Run_0, block-v1:org.0+course_0+Run_0+type@vertical+block@test_unit: 1.0>,
  <Aggregator: Aggregator: robot6, course-v1:org.0+course_0+Run_0, block-v1:org.0+course_0+Run_0+type@vertical+block@test_unit_2: 1.0>,
  <Aggregator: Aggregator: robot6, course-v1:org.0+course_0+Run_0, block-v1:org.0+course_0+Run_0+type@sequential+block@test_subsection: 1.0>,
  <Aggregator: Aggregator: robot6, course-v1:org.0+course_0+Run_0, block-v1:org.0+course_0+Run_0+type@chapter+block@Overview_Chapter: 0.0>,
  <Aggregator: Aggregator: robot6, course-v1:org.0+course_0+Run_0, block-v1:org.0+course_0+Run_0+type@chapter+block@Group_Project: 1.0>,
  <Aggregator: Aggregator: robot6, course-v1:org.0+course_0+Run_0, block-v1:org.0+course_0+Run_0+type@chapter+block@Group_Project2: 1.0>,
  <Aggregator: Aggregator: robot6, course-v1:org.0+course_0+Run_0, block-v1:org.0+course_0+Run_0+type@course+block@course: 0.0>]
```

Note that the earned value for the course aggregator (the last line) is 0.0.  It should be the sum of the chapter aggregators.  I'm looking into why the values aren't bubbling all the way up.

**Comment by jcdyer** [comment]
Note on my above comment: The number on that line is the percent, not the earned.  Those 1.0 values represent 0 earned out of 0 possible.

**Comment by bradenmacdonald** [comment]
@xitij2000 @jcdyer This is missing a review approval and still says "WIP" - can you please clarify the state?

**Comment by xitij2000** [comment]
@bradenmacdonald Since this PR includes a version bump it is WIP since the original PR isn't merged yet. The actual work here is done, but when api-integration is merged, I will update the commit id here.

File: lms/djangoapps/completion_api/models.py [hunk 2]
```diff
@@ -214,6 +213,23 @@ def vertical(self):
...
+        course_metadata = CourseAggregatedMetaData.get_from_id(course_key)
+        total_possible_completions = float(course_metadata.total_assessments)
+        total_actual_completions = StudentProgress.get_total_completions(course_key, exclude_users=exclude_users)
+
+        total_users_qs = CourseEnrollment.objects.users_enrolled_in(course_key).exclude(id__in=exclude_users)
+        total_users = total_users_qs.count()
+        if total_users and total_actual_completions and total_possible_completions:
+            course_avg = total_actual_completions / float(total_users)
+            course_avg = min(100 * (course_avg / total_possible_completions), 100)  # avg in percentage
+        return course_avg
```
Comment (jcdyer) [inline lms/djangoapps/completion_api/models.py]: Is this supposed to still be using the old progress metrics?  (Specifically, `StudentProgress` on L224, and `CourseMdouleCompletion` is still imported on L14). Might be fine.  Just clarifying how this fits in with other outstanding migration PRs.

Comment (jcdyer) [reply lms/djangoapps/completion_api/models.py]: Oh nevermind.  I just saw that this is  in completion_api, which is itself transitional code.  Is this restoring code that was deleted from api_integration?

File: requirements/edx/custom.txt [hunk 3]
```diff
@@ -21,7 +21,7 @@
 -e git+https://github.com/mckinseyacademy/xblock-diagnosticfeedback.git@v0.2.4#egg=xblock-diagnostic-feedback==0.2.4
 -e git+https://github.com/open-craft/xblock-group-project-v2.git@0.4.9#egg=xblock-group-project-v2==0.4.9
 -e git+https://github.com/open-craft/xblock-virtualreality.git@v0.1.1#egg=xblock-virtualreality==0.1.1
-git+https://github.com/edx-solutions/api-integration.git@v1.8.5#egg=api-integration==1.8.5
+git+https://github.com/edx-solutions/api-integration.git@kshitij/port-completion-v1#egg=api-integration==2.0.0
```
Comment (jcdyer) [inline requirements/edx/custom.txt]: :+1: pending the version bump here when that lands on api-integration.

Comment (jcdyer) [reply requirements/edx/custom.txt]: Also pending resolution of the failing tests, of course.

Comment (xitij2000) [reply requirements/edx/custom.txt]: I got all the tests to pass, these seem to have popped up post rebase, so not sure what's gone wrong. Will look into it.

Comment (jcdyer) [reply requirements/edx/custom.txt]: Thanks.  Not sure I understand either.  It looks clean.

File: lms/djangoapps/completion_api/models.py (hunk 2 above)
Comment (xitij2000) [reply lms/djangoapps/completion_api/models.py]: Yes, this was importing code for calculating means from api-integration. That code is now adapted to work with the new completion API so it no longer works with the old API. So I've moved the original code here.

File: requirements/edx/custom.txt (hunk 3 above)
Comment (xitij2000) [reply requirements/edx/custom.txt]: I think I messed up somewhere and a change I had reverted got reintroduced here. I've fixed it and now the tests are passing with the [latest commit](https://github.com/edx-solutions/api-integration/pull/138/commits/3e65826ac18609fa1edb5f51aac3320e0a1b3860)

**Comment by xitij2000** [comment]
@jcdyer Tests are passing now.

**Comment by jcdyer** [comment]
Rebased onto latest development branch.  api-integration version bumped to v2.0.0

**Comment by aamishbaloch** [comment]
@jcdyer Any estimate on when this PR can be merged after fixing the failing tests?

**Comment by xitij2000** [comment]
@aamishbaloch It's mainly just the failing tests that I am trying to figure out, once those are fixed I don't see any delay.

File: requirements/edx/custom.txt [hunk 4]
```diff
@@ -22,13 +22,14 @@
...
+git+https://github.com/edx-solutions/api-integration.git@v2.0.0#egg=api-integration==2.0.0
 git+https://github.com/edx-solutions/organizations-edx-platform-extensions.git@v1.2.1#egg=organizations-edx-platform-extensions==1.2.1
 git+https://github.com/edx-solutions/gradebook-edx-platform-extensions.git@1.1.9#egg=gradebook-edx-platform-extensions==1.1.9
 git+https://github.com/edx-solutions/projects-edx-platform-extensions.git@v1.1.6#egg=projects-edx-platform-extensions==1.1.6
 git+https://github.com/edx-solutions/discussion-edx-platform-extensions.git@v1.1.9#egg=discussion-edx-platform-extensions==1.1.9
 git+https://github.com/edx-solutions/course-edx-platform-extensions.git@v1.1.1#egg=course-edx-platform-extensions==1.1.1
 git+https://github.com/edx-solutions/mobileapps-edx-platform-extensions.git@v1.2.2#egg=mobileapps-edx-platform-extensions==1.2.2
 git+https://github.com/edx-solutions/progress-edx-platform-extensions.git@1.0.8#egg=progress-edx-platform-extensions==1.0.8
-openedx-completion-aggregator==1.2
+git+https://github.com/edx-solutions/progress-edx-platform-extensions.git@1.0.8#egg=progress-edx-platform-extensions==1.0.8
```
Comment (jcdyer) [inline requirements/edx/custom.txt]: This is duplicated from L32.  Please remove one copy.

**Pull request merged**